crates/turbopack-core/src/resolve/pattern.rs

```diff
@@ -764,12 +764,29 @@ impl ValueToString for Pattern {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Clone, PartialOrd, Ord, TraceRawVcs, Serialize, Deserialize)]
+#[derive(Debug, PartialEq, Eq, Clone, TraceRawVcs, Serialize, Deserialize)]
 pub enum PatternMatch {
     File(RcStr, Vc<FileSystemPath>),
     Directory(RcStr, Vc<FileSystemPath>),
 }
 
+impl PartialOrd<PatternMatch> for PatternMatch {
+    fn partial_cmp(&self, other: &PatternMatch) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for PatternMatch {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match (self, other) {
+            (PatternMatch::File(a, _), PatternMatch::File(b, _))
+            | (PatternMatch::Directory(a, _), PatternMatch::Directory(b, _)) => a.cmp(b),
```

Comment:
This may disagree with the `Eq` implementation. The rustdocs aren't very explicit about the relationship between `Eq` and `Ord`, but I think this may violate the requirements for implementing `Ord`?

If we only need `PartialOrd`, I think we should go ahead and implement that, but leave off `Ord`.

Reply:
If you are worried, we could omit PartialOrd and Ord and use `sort_by_key` instead